scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,314 @@
+        elems_iter = struct.iter_unpack('IHBBII', elems_value)
+        i = 0
+
+        for elem in elems_iter:
+            i += 1
+            if i > elems_count:
+                print("Extracted more elems than \'elem_count\'")
+                sys.exit(1)
+            __rt, loc, sig_count, vnd_count, sig_ptr, vnd_ptr = elem
+            elem_item = comp.elem_add(loc)
+
+            def models_unpack(ptr, elem_item, vnd):
+                models_value = read_symbol_data(elf, ptr)
+                # FIXME: Use format depending on Kconfiguration
+                model_format = 'HHIIIHHIHH' + ('I' if label_cnt > 0  else '') + 'II' + ('I' if lcd_srv else '')
+                models_iter = struct.iter_unpack(model_format, models_value)
+
+                for model in models_iter:
+                    id1, id2, __rt, __pub, __keys, __keys_cnt, _, __groups, __groups_cnt, _, __uuids, __op, __cb = model
+                    if not vnd:
+                        elem_item.sig_model_add(id1)
+                    else:
+                        elem_item.vnd_model_add(id1, id2)
+
+            if sig_count > 0:
+                models_unpack(sig_ptr, elem_item, False)
+            if vnd_count > 0:
+                models_unpack(vnd_ptr, elem_item, True)
+
+def version_parse(cfg):
+    if 'CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION' not in cfg.keys():
+        print("Image tool sign version is missing!")
+        return None
+
+    clean_str = cfg['CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION'].replace("+", ".").replace("\"", "")
+    version_list = [int(s) for s in clean_str.split(".") if s.isdigit()]
+    return {
+        "major": version_list[0],
+        "minor": version_list[1],
+        "revision": version_list[2],
+        "build_number": version_list[3],
+    }
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("Missing project folder argument")
+        sys.exit(1)
+
+    bin_path = sys.argv[1]
+    zip_file_name = 'mesh_metadata.json'
+    zip_path = os.path.abspath(bin_path + '/dfu_application.zip')
+    file_name = 'dfu_application.zip_mesh_metadata.json'
+    metadata_path = os.path.abspath(bin_path + '/' + file_name)
+    elf_path = os.path.abspath(bin_path + '/zephyr.elf')
+    config_path = os.path.abspath(bin_path + '/.config')
+
+
+    zip = ZipFile(zip_path, "a")
+    if zip_file_name in zip.namelist():
+        print("Mesh metadata already present in zip file")
+        sys.exit(0)
+
+    comp = Comp0()
+    kconfigs = parse_kconfig(config_path)
+    comp.features_add(kconfigs)
+    addrs = find_comp_data_from_dwarf(elf_path)
+    for addr in addrs:
+        read_comp_data(elf_path, addr, kconfigs, comp)
+
+    j_dict = {
+        "sign_version": version_parse(kconfigs),
+        "composition_hash": str(hex(comp.hash_generate())),
+        "composition_data": comp.dict_generate(),
+    }
+
+    with open(metadata_path, "w") as outfile:
+        outfile.write(json.dumps(j_dict, indent=4))
```

Comment:
Could you also print the hash and the metadata into console? That would make it easy for users to do manual commands by copy-pasting metadata directly from console instead of having to extract the zip and manifest etc.


```
Hash: 0937465829
Metadata: 0010303030040494954865755757577365622
...
...
```

Reply:
Well, they do not need to extract it. It is also located outside the zip as dfu_application.zip_mesh_metadata.json, just like for the existing manifest. I can add a print for the Hash, but I would like to avoid printing too much since it kind of polutes the build output if you just pile on line after line of information.